File: webapp.client/src/app/services/entities/product.ts
interface IProduct {
    id: string;
    name: string | null;
    description: string | null;
    price: number;
    enable: boolean;
    createDate: Date;
}

export class Product implements IProduct {
    id: string;
    name: string | null;
    description: string | null;
    price: number;
    enable: boolean;
    createDate: Date;

    constructor(id: string, name: string | null = null, description: string | null = null, price: number = 0, enable: boolean = true, createDate: Date = new Date()) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.enable = enable;
        this.createDate = createDate;
    }
}

File: WebApp.Domain/Entities/Product.cs
using Dsr.Architecture.Domain.Entities;
using Dsr.Architecture.Utilities;

using WebApp.Domain.Interfaces.Domain.Entities;

namespace WebApp.Domain.Entities;

/// <summary>
/// Represents a product in the inventory system.
/// </summary>
public class Product : Entity<Guid>, IProduct 
{
    /// <summary>
    /// Gets or sets the name of the product.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the description of the product.
    /// </summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the price of the product.
    /// </summary>
    public decimal Price { get; set; } 
    
    /// <summary>
    /// Initializes a new instance of the <see cref="Product"/> class with default values.
    /// </summary>
    public Product() : base(Guid.NewGuid())
    {
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="Product"/> class with a specified ID.
    /// </summary>
    /// <param name="id"></param>
    public Product(Guid? id) : base(id ?? Guid.NewGuid())
    {
    }
    
    /// <summary>
    /// Initializes a new instance of the <see cref="Product"/> class with default values.
    /// </summary>
    /// <param name="name"></param>
    /// <param name="description"></param>
    /// <param name="price"></param>
    /// <param name="id"></param>
    public Product(string name, string description, decimal price, Guid? id = null) : this(id)
        => (Name, Description, Price) = (name, description, price);

}